### ISSUE TITLE
Wrong share messages in the Activities

### ISSUE THREAD

**Issue by cdamken** [owncloud/activity#613]
I saw in the activities from the sync-client and the Activity app:

**You shared oC-Install/index.html with Christian XXXX, Christian XXXX and Dmitry YYYY**

I only shared with Christian and I shared with Dmitry a different File (in different Time)

I can see in the Logs only this related line:

```
{"reqId":"H85WNLSMht5y6TQJZJgC","level":1,"time":"2018-03-20T15:41:32+01:00","remoteAddr":"87.148.35.15","user":"carlos","app":"admin_audit","method":"POST","url":"\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/shares?format=json","message":"user <carlos>, displayName <Carlos Damken> shared file \/carlos\/files\/oC-Install\/index.html with the user chris, permissions:  READ  UPDATE  SHARE, owner: carlos [CLIENT_IP: 87.148.35.15] [USER_AGENT: Mozilla\/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/65.0.3325.162 Safari\/537.36]"}
```

I never shared this file with Dmitry, what can be done? where can I search for more information?

**Comment by PVince81** [comment]
### Steps:

1. Create two users "user1" and "user2"
2. Create two folders "one" and "two"
3. Share "one" with "user1"
4. Share "two" with "user2"
5. Go to activity page

### Expected result
Two entries:
- "You shared 'two' with 'user2'"
- "You shared 'one' with 'user1'"

### Actual result
A single wrong entry: "You shared 'two' with 'user2'  and 'user1'"

### Versions
OC v10.0.8 + activity v2.3.6 (also happening in v2.3.4)
Also observed on latest core stable10 branch and activity master branch.

**Comment by PVince81** [comment]
The "oc_activity" table contains separate entries for both actions
The OCS API delivers a grouped version of these entries.

Seems something is missing in the GroupHelper class that decides on what fields to group on.

**Comment by PVince81** [comment]
PR here: https://github.com/owncloud/activity/pull/624

**Comment by PVince81** [comment]
Keeping open to discuss acceptance tests @patrickjahns

**Comment by PVince81** [comment]
okay, mixed up with other ticket

**Comment by paurakhsharma** [comment]
Acceptance test verifying the fix at: #673
